### PR TITLE
Foolproof the SSDPReceiver timeout mechanism

### PR DESCRIPTION
**Pull request by ebarnard**
On OS X it is possible to have addresses that cannot receive packets but can send them so set a socket read timeout just to be safe.

I noticed this on tun devices with ipv6 link-local addresses. Previously the `receive_packets` would never time out.

### CODE REVIEW

**Comment by GGist** [comment]
Ah yes. It seems that this would also fix the issue of having ```UdpSockets``` bound to ```0.0.0.0``` not being woken up when sending to their ```local_addr()``` to see that the kill flag has been set.

For the AppVeyor build, it looks like ```rustc 1.12.0``` broke something with the Windows build using unstable features. I am going to install the latest nightly and see about fixing that before testing these changes and merging.

File: src/receiver.rs [hunk 1]
```diff
@@ -195,6 +200,11 @@ fn receive_packets<T>(recv: PacketReceiver, kill: Arc<AtomicBool>, send: Sender<
         trace!("Waiting on packet at {}...", recv);
         let (msg_bytes, addr) = match recv.recv_pckt() {
             Ok((bytes, addr)) => (bytes, addr),
+            Err(ref err) if err.kind() == io::ErrorKind::WouldBlock => {
```
Comment (GGist) [inline src/receiver.rs]: While testing on Windows I found that timeouts do not return ```WouldBlock``` but instead ```TimedOut```. Is this behavior consistent on Mac OS as well?

Comment (ebarnard) [inline src/receiver.rs]: OS X and Linux return `WouldBlock` (see: https://is.gd/TFOY1G). There should probably be a rust issue for this.

I suppose for the moment the check should be for both.

**Comment by ebarnard** [comment]
Looks like there was: https://github.com/rust-lang/rust/issues/27773#issuecomment-131196243
And in fact it's mentioned and documented: https://doc.rust-lang.org/stable/std/net/struct.UdpSocket.html#method.set_read_timeout

File: src/receiver.rs (hunk 1 above)
Comment (ebarnard) [inline src/receiver.rs]: Looks like there was: https://github.com/rust-lang/rust/issues/27773#issuecomment-131196243
And in fact it's mentioned and documented: https://doc.rust-lang.org/stable/std/net/struct.UdpSocket.html#method.set_read_timeout

**Comment by GGist** [comment]
Ah yes, I didn't think to check the docs first; I think their decision in the linked issue seems reasonable.

Anyway, all is looking good, thanks for the pull request!

Cheers!

**Pull request merged**